packages/legend-shared/src/CommonUtils.ts

```diff
@@ -224,6 +224,14 @@ export const shallowStringify = (object: unknown): string =>
       : val,
   );
 
+export const validateValidString = (input: string): boolean => {
```

Comment:
we can do one-liner here

Reply:
let's rename this to `isValidIdentifier` and it could be in `formatterUtils` or probably even better place is `@finos/legend-graph/src/MetaModelUtil`, potentially, this can take a flag to accept `$` or not